File: message_passing/echo/erlang/echo.erl
-module(echo).
-export([go/1, loop/0]).

go(Message) ->
  Pid = spawn(echo, loop, []),
  Pid ! {self(), Message},
  receive
    {Pid, Message} ->
      io:format("~s~n", [Message])
  end,
  Pid ! stop.

loop() ->
  receive
    {From, Message} ->
      From ! {self(), Message},
      loop();
    stop ->
      ok
  end.

File: message_passing/echo/elixir/echo.ex
defmodule Echo do

  def go(message) do

    # @doc "burn Pid Process" #1
    pid = spawn(Echo, :loop, [])

    # @doc "send message from self" #2
    send pid, {message, self()}

    receive do
      #5
      msg -> IO.puts("#{msg}")
    end

   # BOTH same func which is shown's message of stack from self() process
     # :erlang.process_info self(), :messages
     # Process.info self(), :messages

  end

  def loop do
    # @doc "pid receive message of self" #3
    receive do
      #  @doc "resend message from self" #4
      {msg, from_pid} ->
        send from_pid, msg
    end
  end
end

